detection-rules/impersonation_openai.yml

```
@@ -0,0 +1,110 @@
+name: "Brand Impersonation: OpenAI"
+description: "Detects messages impersonating OpenAI through display name spoofing or lookalike domains, while excluding legitimate OpenAI communications and trusted senders with valid DMARC authentication."
+type: "rule"
+severity: "medium"
+source: |
+  type.inbound
+  and (
+    // Brand impersonation indicators
+    strings.icontains(sender.display_name, "openai")
+    or strings.icontains(sender.display_name, "chatgpt")
+    or any(ml.logo_detect(file.message_screenshot()).brands,
+           .name == "OpenAI" and .confidence not in ("low")
+    )
+    // OpenAI copyright footer
+    or (
+      strings.icontains(body.current_thread.text, '© 2025 OpenAI, L.L.C.')
+      and (
+        strings.icontains(body.current_thread.text, '3180 18th Street')
+        and strings.icontains(body.current_thread.text, 'San Francisco')
+        and strings.icontains(body.current_thread.text, 'CA 94110')
+
+      )
+    )
+    // Typosquatting domains in links
+    or any(body.links,
+           0 < strings.ilevenshtein(.display_url.domain.root_domain, "openai.com") < 3
+           and .href_url.domain.root_domain != "openai.com"
+    )
+    or any(body.links,
+           0 < strings.ilevenshtein(.display_url.domain.root_domain, "chatgpt.com") < 3
+           and .href_url.domain.root_domain != "chatgpt.com"
+    )
+  )
+  // Exclude observed legit domains
+  and sender.email.domain.root_domain not in~ (
+    "openai.com",
+    "chatgpt.com",
+    "andopen.co",
+    "openli.com",
+    "discoursemail.com",
+    "crunchbase.com",
+    "safebase.co",
+    "lucid.co",
+    "astronomer.io",
+    "visitly.io",
+    "rocketmoney.com"
+  )
+  // Bypass trust/reputation negations for hosting platforms (always suspicious)
+  // Otherwise apply standard high-trust and top-domain negations
+  and (
+    sender.email.domain.root_domain in~ (
+      "firebaseapp.com",
+      "web.app",
+      "appspot.com",
+      "herokuapp.com",
+      "vercel.app",
+      "netlify.app",
+      "github.io",
+      "cloudflareaccess.com",
+      "pages.dev",
+      "repl.co"
+    )
+    or (
+      (
+        sender.email.domain.root_domain not in $high_trust_sender_root_domains
+        or not headers.auth_summary.dmarc.pass
+      )
+      and sender.email.domain.root_domain not in $tranco_1m
+    )
+  )
```

i think something like this then?
```suggestion
  and (
   // not a high trust sender root domain
    not (
      sender.email.domain.root_domain not in $high_trust_sender_root_domains
      and coalesce(headers.auth_summary.dmarc.pass, false)
    )
    // or a hosting platforms (always suspicious) which might be in high trust
    or sender.email.domain.root_domain in~ (
      "firebaseapp.com",
      "web.app",
      "appspot.com",
      "herokuapp.com",
      "vercel.app",
      "netlify.app",
      "github.io",
      "cloudflareaccess.com",
      "pages.dev",
      "repl.co"
    )
  )
```